src/magic_enchantment.cpp

```diff
@@ -0,0 +1,453 @@
+            return true;
+        } else {
+            return false;
+        }
+    }
+    return false;
+}
+
+void enchantment::add_activation( const time_duration &dur, const fake_spell &fake )
+{
+    intermittent_activation[dur].emplace_back( fake );
+}
+
+void enchantment::load( JsonObject &jo, const std::string & )
+{
+    optional( jo, was_loaded, "id", id, enchantment_id( "" ) );
+    if( jo.has_array( "hit_you_effect" ) ) {
+        JsonArray jarray = jo.get_array( "hit_you_effect" );
+        while( jarray.has_more() ) {
+            fake_spell fake;
+            JsonObject fake_spell_obj = jarray.next_object();
+            fake.load( fake_spell_obj );
+            hit_you_effect.emplace_back( fake );
+        }
+    }
+
+    if( jo.has_array( "hit_me_effect" ) ) {
+        JsonArray jarray = jo.get_array( "hit_me_effect" );
+        while( jarray.has_more() ) {
+            fake_spell fake;
+            JsonObject fake_spell_obj = jarray.next_object();
+            fake.load( fake_spell_obj );
+            hit_me_effect.emplace_back( fake );
+        }
+    }
+
+    if( jo.has_object( "intermittent_activation" ) ) {
+        JsonObject jobj = jo.get_object( "intermittent_activation" );
+        JsonArray jarray = jo.get_array( "effects" );
+        while( jarray.has_more() ) {
+            JsonObject effect_obj;
+            time_duration dur = read_from_json_string<time_duration>( *effect_obj.get_raw( "frequency" ),
+                                time_duration::units );
+            if( effect_obj.has_array( "spell_effects" ) ) {
+                JsonArray jarray = effect_obj.get_array( "spell_effects" );
+                while( jarray.has_more() ) {
+                    fake_spell fake;
+                    JsonObject fake_spell_obj = jarray.next_object();
+                    fake.load( fake_spell_obj );
+                    add_activation( dur, fake );
+                }
+            } else if( effect_obj.has_object( "spell_effects" ) ) {
+                fake_spell fake;
+                JsonObject fake_spell_obj = effect_obj.get_object( "spell_effects" );
+                fake.load( fake_spell_obj );
+                add_activation( dur, fake );
+            }
+        }
+    }
+
+    active_conditions.first = io::string_to_enum<has>( jo.get_string( "has", "HELD" ) );
+    active_conditions.second = io::string_to_enum<condition>( jo.get_string( "condition",
+                               "ALWAYS" ) );
+
+    if( jo.has_array( "values" ) ) {
+        JsonArray jarray = jo.get_array( "values" );
+        while( jarray.has_more() ) {
+            JsonObject value_obj = jarray.next_object();
+            const enchantment::mod value = io::string_to_enum<mod>( value_obj.get_string( "value" ) );
+            const int add = value_obj.get_int( "add", 0 );
+            const double mult = value_obj.get_float( "multiply", 0.0 );
+            if( add != 0 ) {
+                values_add.emplace( value, add );
```

Comment:
If #33742 is merged then `values_add` and `values_multiply` should be possible to (de)serialize in a single line, rather than rolling your own loops.  The structure of the resulting json would be different, though.